core/src/main/scala/org/apache/spark/sql/execution/columnar/JDBCAppendableRelation.scala

```diff
@@ -66,9 +66,23 @@ abstract case class JDBCAppendableRelation(
     externalStore.connProperties
 
   protected final val connFactory: () => Connection = JdbcUtils
-      .createConnectionFactory(new JDBCOptions(connProperties.url,
+      .createConnectionFactory(new JDBCOptions(connProperties.url +
+        connProperties.urlSecureSuffix,
         table, connProperties.connProps.asScala.toMap))
 
+  protected final val sysConnFactory: () => Connection = {
+    val EMPTY_URL = ""
+    val user = sqlContext.conf.getConfString("snappydata.store.user", EMPTY_URL)
+    val password = sqlContext.conf.getConfString("snappydata.store.password", EMPTY_URL)
+    if (!user.equals(EMPTY_URL) && !password.equals(EMPTY_URL)) {
+      JdbcUtils.createConnectionFactory(new JDBCOptions(connProperties.url +
+        ";user=" + user +
+        ";password=" + password +
+        ";default-schema=" + SnappyStoreHiveCatalog.HIVE_METASTORE + ";",
+        table, connProperties.connProps.asScala.toMap))
+    } else connFactory
```

Comment:
This is being done in multiple places. Refactor this portion out into a separate method in Utils or somewhere that will read the properties from SQLConf and fill into connProperties (preferably into connProps and executorConnProps as mentioned before).